src/Compilers/CSharp/Portable/Errors/MessageID.cs

```diff
@@ -230,12 +230,30 @@ internal static string RequiredFeature(this MessageID feature)
             }
         }
 
-        internal static bool CheckFeatureAvailability(this MessageID feature, DiagnosticBag diagnostics, Location errorLocation)
+        internal static bool CheckFeatureAvailability(
+            this MessageID feature,
+            DiagnosticBag diagnostics,
+            SyntaxNode syntax,
+            Location location = null)
         {
-            var diag = GetFeatureAvailabilityDiagnosticInfoOpt(feature, (CSharpParseOptions)errorLocation.SourceTree.Options);
+            var diag = GetFeatureAvailabilityDiagnosticInfoOpt(feature, (CSharpParseOptions)syntax.SyntaxTree.Options);
             if (!(diag is null))
```

Comment:
>!(diag is null) [](start = 16, length = 15)

Nit: while you're here, consider simplifying to `diag is object`

Reply:
The variable declaration could also be inlined entirely.

---
In reply to: [308427113](https://github.com/dotnet/roslyn/pull/37513#discussion_r308427113) [](ancestors = 308427113)

Reply:
True, but it's not bad as-is.